AlfrescoSDK/AlfrescoSDK/Session/OnPremiseSession/AlfrescoRepositorySession.m

```diff
@@ -72,14 +65,35 @@ + (AlfrescoRequest *)connectWithUrl:(NSURL *)url
                          parameters:(NSDictionary *)parameters
                     completionBlock:(AlfrescoSessionCompletionBlock)completionBlock
 {
-    [AlfrescoErrors assertArgumentNotNil:url argumentName:@"url"];
-    [AlfrescoErrors assertArgumentNotNil:username argumentName:@"username"];
-    [AlfrescoErrors assertArgumentNotNil:password argumentName:@"password"];
-    [AlfrescoErrors assertArgumentNotNil:completionBlock argumentName:@"completionBlock"];
-    AlfrescoRepositorySession *sessionInstance = [[AlfrescoRepositorySession alloc] initWithUrl:url parameters:parameters];
-    if (sessionInstance) 
+    NSDictionary *startDict = [AlfrescoConnectionDiagnosticUtils createDictionaryForStartEventForEventName:kAlfrescoConfigurationDiagnosticReachabilityEvent];
+    [[NSNotificationCenter defaultCenter] postNotificationName:kAlfrescoConfigurationDiagnosticDidStartEventNotification object:nil userInfo:startDict];
```

Comment:
Agreed - it's an app responsibility to ensure UIKit methods are called on the main thread. This isn't a job for the SDK.